go/vt/servenv/servenv.go

```diff
@@ -351,6 +351,17 @@ func ParseFlags(cmd string) {
 	logutil.PurgeLogs()
 }
 
+// ParseFlagsForUnitTests initializes flags but skips the filesystem args
+// and go flag related work.
+// Note: this should NOT be used outside of unit tests.
```

Comment:
echoing what i've said elsewhere: if you want to restructure things a bit with an internal package, you can make this impossible to call from not-a-test

named the files for the functions that they’d contain, because i can’t think of a more concise way to do this textually:

```
.
└── servenv
    ├── internal
    │   └── viper (imports nothing)
    │       └── load.go
    ├── servenv.go (imports viper.Load())
    └── testenv (imports servenv.GetFlagsFor() and viper.Load())
        └── parse_flags_for_test.go
```

Reply:
If not for the fact that this pattern of Foo() and FooForTest[s]() being paired together in the same file with a note for those who are somehow fooled by the function name.

It's not a bad refactor if there becomes a compelling reason at some point or our backlog gets that thin, but IMO it's unnecessary overkill just for this case alone.